enterprise/coderd/appearance_test.go

```diff
@@ -11,68 +11,125 @@ import (
-	require.NoError(t, err)
-	require.False(t, sb.ServiceBanner.Enabled)
-
-	basicUserClient, _ := coderdtest.CreateAnotherUser(t, adminClient, adminUser.OrganizationID)
-
-	uac := codersdk.UpdateAppearanceConfig{
-		ServiceBanner: sb.ServiceBanner,
-	}
-	// Regular user should be unable to set the banner
-	uac.ServiceBanner.Enabled = true
-
-	err = basicUserClient.UpdateAppearance(ctx, uac)
-	require.Error(t, err)
-	var sdkError *codersdk.Error
-	require.True(t, errors.As(err, &sdkError))
-	require.Equal(t, http.StatusForbidden, sdkError.StatusCode())
-
-	// But an admin can
-	wantBanner := uac
-	wantBanner.ServiceBanner.Enabled = true
-	wantBanner.ServiceBanner.Message = "Hey"
-	wantBanner.ServiceBanner.BackgroundColor = "#00FF00"
-	err = adminClient.UpdateAppearance(ctx, wantBanner)
-	require.NoError(t, err)
-	gotBanner, err := adminClient.Appearance(ctx)
-	require.NoError(t, err)
-	gotBanner.SupportLinks = nil // clean "support links" before comparison
-	require.Equal(t, wantBanner.ServiceBanner, gotBanner.ServiceBanner)
-
-	// But even an admin can't give a bad color
-	wantBanner.ServiceBanner.BackgroundColor = "#bad color"
-	err = adminClient.UpdateAppearance(ctx, wantBanner)
-	require.Error(t, err)
+	t.Run("Agent", func(t *testing.T) {
+		t.Parallel()
+
+		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
+		defer cancel()
+
+		client := coderdenttest.New(t, &coderdenttest.Options{
+			Options: &coderdtest.Options{
+				IncludeProvisionerDaemon: true,
+			},
+		})
+		user := coderdtest.CreateFirstUser(t, client)
+		license := coderdenttest.AddLicense(t, client, coderdenttest.LicenseOptions{
+			Features: license.Features{
+				codersdk.FeatureAppearance: 1,
+			},
+		})
+		cfg := codersdk.UpdateAppearanceConfig{
+			ServiceBanner: codersdk.ServiceBannerConfig{
+				Enabled:         true,
+				Message:         "Hey",
+				BackgroundColor: "#00FF00",
+			},
+		}
+		err := client.UpdateAppearance(ctx, cfg)
+		require.NoError(t, err)
+
+		authToken := uuid.NewString()
+		agentClient := agentsdk.New(client.URL)
+		agentClient.SetSessionToken(authToken)
+		version := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, &echo.Responses{
+			Parse:          echo.ParseComplete,
+			ProvisionPlan:  echo.ProvisionComplete,
+			ProvisionApply: echo.ProvisionApplyWithAgent(authToken),
+		})
+		template := coderdtest.CreateTemplate(t, client, user.OrganizationID, version.ID)
+		coderdtest.AwaitTemplateVersionJob(t, client, version.ID)
+		workspace := coderdtest.CreateWorkspace(t, client, user.OrganizationID, template.ID)
+		coderdtest.AwaitWorkspaceBuildJob(t, client, workspace.LatestBuild.ID)
+
+		banner, err := agentClient.GetServiceBanner(ctx)
+		require.NoError(t, err)
+		require.Equal(t, cfg.ServiceBanner, banner)
+
+		client.DeleteLicense(ctx, license.ID)
+		banner, err = agentClient.GetServiceBanner(ctx)
+		require.NoError(t, err)
+		require.Equal(t, codersdk.ServiceBannerConfig{}, banner)
+	})
```

Comment:
Since we have two optionals, should we add a test for when both are missing and it's expected behavior?

Reply:
By the two optionals do you mean the status code checks?  I wanted to check the 404 but I was not sure how to run the test with the enterprise routes disabled.  Although now that I look at it a bit more, I think `coderdenttest` is for enterprise and `coderdtest` is non-enterprise?  I will try that.

Or do you mean a test for when the banner is missing and there is no license?

Reply:
Turns out this is going away as I move the banner to be sent along with the other metadata anyway!

Reply:
Er, nevermind, it is not going away just moving to another metadata endpoint.